ui/tests/acceptance/sync/secrets/overview-test.js

```diff
@@ -39,4 +40,38 @@ module('Acceptance | sync | destination', function (hooks) {
     await click(ts.overview.table.action('details'));
     assert.dom(ts.tab('Secrets')).hasClass('active', 'Navigates to secrets view for destination');
   });
+
+  test('it should show opt-in banner and modal if secrets-sync is not activated', async function (assert) {
+    assert.expect(6);
+    server.get('/sys/activation-flags', () => {
+      assert.ok(true, 'Request on initial load to check if secrets-sync is activated');
+      return {
+        data: {
+          activated: [''],
+          unactivated: ['secrets-sync'],
+        },
+      };
+    });
+    await click(ts.navLink('Secrets Sync'));
+    assert.dom(ts.overview.optInBanner).exists('Opt-in banner is shown');
+    await click(ts.overview.optInBannerEnable);
+    assert.dom(ts.overview.optInModal).exists('Opt-in modal is shown');
+    assert.dom(ts.overview.optInConfirm).isDisabled('Confirm button is disabled when checkbox is unchecked');
+    this.server.post('/sys/activation-flags/secrets-sync/activate', () => {
```

Comment:
Can this be moved up so it's next to the other endpoint?

Reply:
Sure can. When I'm writing test, I like to put the endpoint I'm hitting right before the action. however, I know the standard is to put them all at the top.